File: frontend/src/components/Menu/MenuItem.js
import React from "react";
import PriceTag from "../../elements/Tags/PriceTag";
import StockTag from "../../elements/Tags/StockTag";
import { addToCart } from "../../controllers/cartController";
import useUserContext from "../../hooks/useUserContext";
import useCartContext from "../../hooks/useCartContext";

const MenuItem = (({ img, name, price, quantity, selected, id, type }) => {
  const { user } = useUserContext();
  const { cart, dispatch } = useCartContext();
  return (
    <div
      onClick={() => {
        console.log("id:" + id + "FoodItem clicked");
        addToCart(user.token, id)
          .then((res) => {
            if (cart.find((item) => item.id === id)) {
              dispatch({ type: "INCREMENT_ITEM", payload: id });
            } else {
              dispatch({
                type: "ADD_ITEM",
                payload: {
                  icon: img,
                  id: id,
                  name: name,
                  price: price,
                  quantity: 1,
                },
              });
            }
            console.log(res);
          })
          .catch((err) => {
            console.log(err);
          });
      }}
      className={
        "FoodItem h-fit max-w-40 m-2 relative rounded-lg bg-white shadow" +
        (selected ? " border-[3px] border-amber-500 shadow-md" : "")
      }
    >
      <div className="absolute top-[-8px] left-[-8px]">
        <PriceTag price={price} />
      </div>
      <div className="absolute top-[-5px] right-[-8px]">
        <StockTag quantity={quantity} />
      </div>

      <div className="rounded-t-lg overflow-hidden">
        <img
          src={img}
          alt=""
          className="object-cente object-cover aspect-video"
        />
      </div>
      <h3 className="text-gray-600 font-semibold text-center">{name}</h3>
    </div>
  );
});

export default MenuItem;

File: src/src/components/Menu/menu_item.py
import reflex as rx
from src.components.Menu.price_tag import PriceTag
from src.components.Menu.stock_tag import StockTag
# state
from src.state import MenuItemState


def MenuItem(item) -> rx.Component:
    return rx.chakra.box(
        rx.chakra.box(
            PriceTag(item["item_price"]), class_name="absolute top-[-8px] left-[-8px]"
        ),
        rx.chakra.box(
            StockTag(item["item_quantity"]), class_name="absolute top-[-5px] right-[-8px]"
        ),
        rx.chakra.box(
            rx.chakra.image(
                src=item["item_icon"],
                class_name="object-cente object-cover aspect-video",
            ),
            class_name="rounded-t-lg overflow-hidden",
        ),
        rx.chakra.text(item["item_name"], class_name="text-gray-600 font-semibold text-center"),
        class_name=rx.cond(
            MenuItemState.selected,
            "FoodItem h-fit m-2 relative rounded-lg bg-white shadow border-[3px] border-amber-500 shadow-md",
            "FoodItem h-fit m-2 relative rounded-lg bg-white shadow border-[3px] hover:border-primary-400 border-white",
        ),
        max_width="10rem",
    )
